src/SurfaceSchemes/surface_conditions.jl

```diff
@@ -0,0 +1,345 @@
+# TODO: Remove this type piracy.
+Base.Broadcast.BroadcastStyle(
+    ::Base.Broadcast.Style{Tuple},
+    ds::DataLayouts.DataStyle,
+) = ds
+
+"""
+    projected_vector_data(::Type{V}, vector, local_geometry)
+
+Projects the given vector onto the axis of V, then extracts the component data.
+The type should correspond to a vector with only one component, i.e., a basis vector.
+"""
+projected_vector_data(::Type{V}, vector, local_geometry) where {V} =
+    V(vector, local_geometry)[1] / unit_basis_vector_data(V, local_geometry)
+
+
+"""
+    update_surface_conditions!(Y, p, t)
+
+Updates the value of `p.sfc_conditions` based on the current state `Y` and time
+`t`. This function will usually be called automatically, but it needs to be
+called manually if `surface_updater` is set to `ManualUpdates`.
+"""
+function update_surface_conditions!(Y, p, t)
+    (; ᶜts, sfc_conditions, sfc_state, params, atmos) = p
+    sfc_local_geometry = Fields.field_values(
+        Fields.level(Fields.local_geometry_field(Y.f), Fields.half),
+    )
+    int_local_geometry =
+        Fields.field_values(Fields.level(Fields.local_geometry_field(Y.c), 1))
+    int_ts = Fields.field_values(Fields.level(ᶜts, 1))
+    int_u = Fields.field_values(Fields.level(p.ᶜu, 1))
+    int_z = Fields.field_values(Fields.level(Fields.coordinate_field(Y.c).z, 1))
+    sfc_conditions_values = Fields.field_values(sfc_conditions)
+
+    sfc_state_broadcast = if isa(sfc_state, Function)
+        (sfc_state(sfc_local_geometry.coordinates, int_z, t),)
```

Comment:
```suggestion
        Base.Broadcasted(sfc_state, sfc_local_geometry.coordinates, int_z, t)
```
If we don't broadcast this, it's going to throw an error.